src/lib.rs

```diff
@@ -98,6 +97,65 @@ impl<Data: AsRef<[u8]>> Reader<Data> {
             .unwrap()
             .map(move |level| &self.input.as_ref()[level.offset as usize..(level.offset + level.length_bytes) as usize])
     }
+
+    pub fn supercompression_global_data(&self) -> &[u8] {
+        let header = self.header();
+        let start = header.sgd_byte_offset as usize;
+        let end = (header.sgd_byte_offset + header.sgd_byte_length) as usize;
+        &self.input.as_ref()[start..end]
+    }
+
+    pub fn data_format_descriptors(&self) -> ParseResult<impl Iterator<Item = DataFormatDescriptor>> {
```

Comment:
Since we can easily validate this consistently in advance, I'd prefer to do so in `new`, then have the public method be infallible.

Reply:
Done.